train_segmentation.py

```diff
@@ -146,9 +147,12 @@ def create_dataloader(data_path, batch_size, task, num_devices, params, samples_
     num_workers = num_devices * 4 if num_devices > 1 else 4
 
     # Shuffle must be set to True.
-    trn_dataloader = DataLoader(trn_dataset, batch_size=batch_size, num_workers=num_workers, shuffle=True, drop_last=True)
-    val_dataloader = DataLoader(val_dataset, batch_size=batch_size, num_workers=num_workers, shuffle=False)
-    tst_dataloader = DataLoader(tst_dataset, batch_size=batch_size, num_workers=num_workers, shuffle=False) if num_samples['tst'] > 0 else None
+    trn_dataloader = DataLoader(trn_dataset, batch_size=batch_size, num_workers=num_workers, shuffle=True,
+                                drop_last=True)
+    val_dataloader = DataLoader(val_dataset, batch_size=batch_size, num_workers=num_workers, shuffle=False,
+                                drop_last=True)
```

Comment:
We preferred not dropping last items (drop_last=True) for val and test sets since shuffle is False and batch sizes are usually 1 (right?). Therefore, it is not necessary to drop last batch if smaller than batch size. Moreover, it causes for those last samples to be consistently ignored (always the same ones, since shuffle is False). @mpelchat04 That was the reason, right? I think you were the one to bring it up.

I would suggest keeping drop_last = False for val and test dataloaders.

Reply:
The defined batch_size in the config yaml is retained for trn,val and test dataloaders especially when training on GPU. I came across issues when the batch_size divides the number of samples and there is a remainder. This issue does not arise when the batch size occurs in even chunks of the total number of samples. It has nothing to do with the shuffle param.